.storybook/stories/TableFiltering/components/DashboardTable.tsx

```diff
@@ -0,0 +1,262 @@
+	sort?: GetDataSort;
+	/** A function to set the sort */
+	setSort?: (sort: GetDataSort) => void;
+	/** Whether the table is loading */
+	loading?: boolean;
+	/** The total number of items found in the search */
+	totalItems?: number;
+	/** The number of items to display per page */
+	itemsPerPage?: number;
+};
+
+export const DashboardTable = ({
+	caption,
+	data,
+	loading,
+	sort,
+	setSort,
+	itemsPerPage = 10,
+	totalItems,
+}: DashboardTableProps) => {
+	const isTableSortable = !sort || !!setSort;
+
+	if (!loading && data.length === 0) {
+		return <Text>No results</Text>;
+	}
+
+	return (
+		<TableWrapper>
+			<Table aria-rowcount={totalItems}>
+				<TableCaption>
+					{caption}
+					<VisuallyHidden>
+						, column headers with buttons are sortable.
+					</VisuallyHidden>
+				</TableCaption>
+				<TableHead>
+					<tr>
+						{headers.map(
+							({
+								label,
+								sortKey,
+								textAlign,
+								width,
+								isSortable: isFieldSortable,
+							}) => {
+								const isFieldTheActiveSortField = sort?.field === sortKey;
+								const clickHandler = () =>
+									setSort?.({
+										field: sortKey,
+										order:
+											sort?.field === sortKey && sort?.order === 'ASC'
+												? 'DESC'
+												: 'ASC',
+									});
+
+								return (
+									<DashboardTableHeader
+										key={sortKey}
+										textAlign={textAlign}
+										width={width}
+										isSortable={isTableSortable && isFieldSortable}
+										sort={isFieldTheActiveSortField ? sort.order : undefined}
+										onClick={clickHandler}
+									>
+										{label}
+									</DashboardTableHeader>
+								);
+							}
+						)}
+					</tr>
+				</TableHead>
+				<TableBody>
+					{loading ? (
+						<Fragment>
+							{Array.from(Array(itemsPerPage).keys()).map((i) => (
+								<tr key={i}>
+									<TableCell>
+										<SkeletonText />
+										<VisuallyHidden>Loading</VisuallyHidden>
+									</TableCell>
+									<TableCell>
+										<SkeletonText />
+										<VisuallyHidden>Loading</VisuallyHidden>
+									</TableCell>
+									<TableCell>
+										<SkeletonText />
+										<VisuallyHidden>Loading</VisuallyHidden>
+									</TableCell>
+									<TableCell>
+										<SkeletonText />
+										<VisuallyHidden>Loading</VisuallyHidden>
+									</TableCell>
+									<TableCell>
+										<SkeletonText />
+										<VisuallyHidden>Loading</VisuallyHidden>
+									</TableCell>
+									<TableCell>
+										<SkeletonBox height={32} />
+										<VisuallyHidden>Loading</VisuallyHidden>
+									</TableCell>
+								</tr>
+							))}
+						</Fragment>
+					) : (
+						<Fragment>
+							{data.map(
+								({
+									index,
+									id,
+									assignee,
+									businessName,
+									city,
+									state,
+									numberOfEmployees,
+									requestDate,
+									status,
+								}) => {
+									return (
+										<tr key={id} aria-rowindex={index}>
```

Comment:
```suggestion
										<tr key={id} aria-rowindex={index + 1}>
```

Reply:
However, I'm pretty sure that when the data is filtered, it should maintain the rowindex of what it would be when it's unfiltered